### PR TITLE
Move `absolutize_relative_to` into `scolopasta-path`

### LINKED ISSUE #2179: Extract `absolutize_relative_to` to `scolapasta-path`
This path normalization utility lives in `artichoke-backend`.

https://github.com/artichoke/artichoke/blob/2fe83803834963deef45e5fa8a295188902b72b7/artichoke-backend/src/load_path.rs#L66-L246

Move it and its tests to `scolapasta-path`. Does this need more tests?

I'm not convinced that this function is the best way to do this and maybe we can delegate to a third party crate. Some requirements:

- Does not rely on `std::fs::canonicalize` because this routine may be used on paths in virtual filesystems.
- Correctly handles various Windows path prefixes including verbatim paths: https://doc.rust-lang.org/stable/std/path/enum.Prefix.html
- Produces paths with `/` as the path separator (for non-verbatim paths on Windows). This can be done by collecting the `Component`s into an `OsString` with `Component::as_os_str`.

It looks like the current implementation is missing tests that the string representation of the path has `/` path separators on Windows.

### PR DESCRIPTION
**Pull request by NlightNFotis**
Move `absolutize_relative_to` into the `scolopasta-path/` crate, with a small refactor performed as indicated in the original [reporting issue](https://github.com/artichoke/artichoke/issues/2179).

Fixes #2179 (or at least a part of that?).

### CODE REVIEW

**Comment by lopopolo** [comment]
it looks like there is some stray whitespace in this diff. Running `bundle exec rake fmt` should fix things up

File: scolapasta-path/src/lib.rs [hunk 1]
```diff
@@ -34,7 +34,7 @@
 mod paths;
```
Comment (lopopolo) [inline scolapasta-path/src/lib.rs:34]: Could you bump scolapasta-paty to 0.5.0?

You'll have to change its `Cargo.toml`, `README.md`, any direct deps in the various `Cargo.toml`s in the Artichoke workspace, and then run `bundle exec rake build:all` to bump the fuzz and spec-runner lockfiles.

Please do all of these steps in a single, separate commit.

**Comment by NlightNFotis** [comment]
Hello @lopopolo - my apologies for not having gotten back to you earlier on this. I was hosting a friend of mine over the Christmas holiday and a bit into January, which made it very hard for me to engage with any programming work.

I will implement the changes required some time tomorrow (Wednesday) evening, U.K. time.

Thanks for bearing with me and my sincerest apologies for any inconvenience I have caused.

**Comment by lopopolo** [comment]
No worries @NlightNFotis. Your the driver on this one. Take as much time as you need.

**Comment by NlightNFotis** [comment]
Hi @lopopolo, I have updated the PR but they CI checks appear not to have run.

Can you activate them and let me know if there's something more to be done here?

**Review by lopopolo — APPROVED** [review]

**Pull request merged**